File: backend/src/main/java/com/staccato/member/service/dto/response/MemberProfileResponse.java
package com.staccato.member.service.dto.response;

import com.staccato.config.swagger.SwaggerExamples;
import com.staccato.member.domain.Member;
import io.swagger.v3.oas.annotations.media.Schema;


@Schema(description = "프로필을 조회 했을 때 응답 형식입니다.")
public record MemberProfileResponse(
        @Schema(example = SwaggerExamples.MEMBER_NICKNAME)
        String nickname,
        @Schema(example = SwaggerExamples.IMAGE_URL)
        String profileImageUrl,
        @Schema(example = SwaggerExamples.MEMBER_CODE)
        String code) {
    public MemberProfileResponse(Member member) {
        this(member.getNickname().getNickname(), member.getImageUrl(), member.getCode());
    }
}

File: android/Staccato_AN/data/src/main/java/com/on/staccato/data/dto/mypage/MemberProfileResponse.kt
package com.on.staccato.data.dto.mypage

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class MemberProfileResponse(
    @SerialName("memberId") val memberId: Long,
    @SerialName("nickname") val nickname: String,
    @SerialName("profileImageUrl") val profileImageUrl: String? = null,
    @SerialName("code") val code: String,
)
